### PR TITLE
Add note to avoid confusing

### LINKED ISSUE #4456: lint emitting updating
if you don't run `cargo uitest` (and maybe `cargo test`) before `tests/ui/update-all-references.sh` the script won't do anything.

It would be nice if this could be made more clear. Got me really confused why `cargo run` and `cargo build` didn't trigger a change for the script.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #4456 

changelog: none

### CODE REVIEW

File: doc/adding_lints.md [hunk 1]
```diff
@@ -86,6 +86,8 @@ test. That allows us to check if the output is turning into what we want.
 
 Once we are satisfied with the output, we need to run
 `tests/ui/update-all-references.sh` to update the `.stderr` file for our lint.
+Please note that, we should run `TESTNAME=ui/foo_functions cargo uitest`
+before running `tests/ui/update-all-references.sh`.
```
Comment (BO41) [inline doc/adding_lints.md]: maybe add "every time", as this is the frustrating part.
You change something, but the `.stderr` doesn't change until you run `uitest` _again_

Comment (JohnTitor) [reply doc/adding_lints.md]: Thanks! I added "every time".

**Comment by flip1995** [comment]
Thanks!

@bors r+

**Comment by JohnTitor** [comment]
Hmm, the test seems failed due to latest rustc changes. I guess `Arg` is replaced with `Param`.

**Comment by JohnTitor** [comment]
Hopefully fixed in https://github.com/rust-lang/rust-clippy/pull/4462

**Pull request merged**